packages/logging-winston/src/index.js

```diff
@@ -0,0 +1,189 @@
+ *     require you to specify `options.email` as well.
+ * @property {string=} email - Account email address. Required when using a .pem
+ *     or .p12 keyFilename.
+ * @property {object=} credentials - Credentials object.
+ * @property {string} credentials.client_email
+ * @property {string} credentials.private_key
+ * @property {boolean=} autoRetry - Automatically retry requests if the
+ *     response is related to rate limits or certain intermittent server errors.
+ *     We will exponentially backoff subsequent requests by default. (default:
+ *     true)
+ * @property {number=} maxRetries - Maximum number of automatic retries
+ *     attempted before returning the error. (default: 3)
+ * @property {function=} promise - Custom promise module to use instead
+ *     of native Promises.
+ */
+
+/**
+ * Constructs a new object which can be used as a Winston transport.
+ * @param {OptionsObject=} options - optional configuration object.
+ * @constructor
+ */
+var StackdriverTransport =
+    winston.transports.StackdriverLogging = function(options) {
+      options = options || {};
+      var logName = options.logName || 'winston_log';
+
+      var transportOptions = {level: options.level, name: logName};
+      Transport.call(this, transportOptions);
+
+      this.levels_ = options.levels || npmLevels;
+      this.metadata_ = {logName: logName};
+      if (options.resource) {
+        this.metadata_.resource = _.assign({}, options.resource);
+      }
+      var logging = require('@google-cloud/logging')(options);
+      this.logger_ = logging.log(logName);
+    };
+
+util.inherits(StackdriverTransport, Transport);
+
+/**
+ * Relays a log entry to the logging agent. This is normally called by winston.
+ * @param {string} level - The severity level at which this entry is being
+ *     logged. This should match one of the levels provided to the constructor
+ *     which defaults to npm levels. This level will be translated to the
+ *     appropriate Stackdriver logging severity level.
+ * @param {string} msg - The message to be logged.
+ * @param {object=} meta - Winston provided metadata that should be attached to
+ *     log entry. Each property will be converted to a string using
+ *     `util.inspect`.
+ * @param {function=} callback A callback that is invoked when the logging agent
+ *     either succeeds or gives up writing the log entry to the remote server.
+ */
+StackdriverTransport.prototype.log = function(level, msg, meta, callback) {
+  if (_.isFunction(meta) && !callback) {
+    callback = meta;
+    meta = null;
+  }
+
+  if (!_.has(this.levels_, level)) {
+    throw new Error('unknown log level ' + level);
+  }
+
+  // Translate `level` to a Stackdriver logging level.
+  var numericLevel = this.levels_[level];
+  var stackdriverLevel = levelNames[numericLevel];
+
+  var metadata = this.metadata_;
+  if (meta) {
+    // Logging proto requires that the label values be strings, so we convert
+    // using util.inspect.
```

Comment:
Do you think we should use `JSON.stringify` instead of `util.inspect`?:

```js
> JSON.stringify({ hey: 'there' })
'{"hey":"there"}'
> util.inspect({ hey: 'there' })
'{ hey: \'there\' }'
```

That's what we do in other places in the library, but I'm not sure if `util.inspect` is better here.